linalg/auxiliary.cpp

```diff
@@ -0,0 +1,542 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "../config/config.hpp"
+
+#ifdef MFEM_USE_MPI
+
+#include "linalg.hpp"
+#include "../fem/pfespace.hpp"
+#include "../fem/pbilinearform.hpp"
+
+namespace mfem
+{
+
+GeneralAMS::GeneralAMS(const Operator& curlcurl_op_,
+                       const Operator& pi_,
+                       const Operator& gradient_,
+                       const Operator& pispacesolver_,
+                       const Operator& gspacesolver_,
+                       const Operator& smoother_,
+                       const Array<int>& ess_tdof_list_)
+   :
+   Solver(curlcurl_op_.Height()),
+   curlcurl_op(curlcurl_op_),
+   pi(pi_),
+   gradient(gradient_),
+   pispacesolver(pispacesolver_),
+   gspacesolver(gspacesolver_),
+   smoother(smoother_),
+   ess_tdof_list(ess_tdof_list_),
+   residual_time(0.0),
+   smooth_time(0.0),
+   gspacesolver_time(0.0),
+   pispacesolver_time(0.0)
+{
+   // could assert a bunch of sizes...
```

Comment:
Is this comment (and some others of the same kind) intended to stay?

Reply:
I have gone over the comments in this file, removing some and rewriting others to be more professional as hints to future maintainers. There could still probably be improvements, feel free to make more suggestions.